### PR TITLE
[safety-rules] introduce new storage layer [4/n]

### PR DESCRIPTION
**Pull request by davidiw**
Follows https://github.com/libra/libra/pull/2314

This brings in the new, generic secure storage layer into safety rules.
There is an additional commit that performs some cleanup operations in safety rules.

### CODE REVIEW

File: consensus/safety-rules/src/persistent_storage.rs [hunk 1]
```diff
@@ -14,20 +15,27 @@ pub struct PersistentStorage {
...
 }
 
+const CONSENSUS_KEY: &str = "consensus_key";
 const EPOCH: &str = "epoch";
 const LAST_VOTED_ROUND: &str = "last_voted_round";
 const PREFERRED_ROUND: &str = "preferred_round";
 
 impl PersistentStorage {
-    pub fn in_memory() -> Self {
+    pub fn in_memory(private_key: Ed25519PrivateKey) -> Self {
```
[the remark this answers is not in the document]
Comment (davidiw) [reply consensus/safety-rules/src/persistent_storage.rs]: Hahaha, well even in that case, we'd need to change the internal code to handle that agility appropriately.

File: consensus/safety-rules/src/safety_rules.rs [hunk 2]
```diff
@@ -28,19 +34,19 @@ use std::{marker::PhantomData, sync::Arc};
...
 impl<T: Payload> SafetyRules<T> {
     /// Constructs a new instance of SafetyRules with the given persistent storage and the
     /// consensus private keys
     /// @TODO replace this with an API that takes in a SafetyRulesConfig
-    /// @TODO load private key from persistent store
-    pub fn new(
-        persistent_storage: PersistentStorage,
-        validator_signer: Arc<ValidatorSigner>,
-    ) -> Self {
+    pub fn new(author: Author, persistent_storage: PersistentStorage) -> Self {
```
[the remark this answers is not in the document]
Comment (davidiw) [reply consensus/safety-rules/src/safety_rules.rs]: In the interest of avoiding rebase hell, I'm going to ask to defer. Or maybe do this right before I land...

File: consensus/safety-rules/src/persistent_storage.rs [hunk 3]
```diff
@@ -3,213 +3,128 @@
...
+
+impl PersistentStorage {
+    pub fn in_memory(private_key: Ed25519PrivateKey) -> Self {
+        let storage = Box::new(InMemoryStorage::new());
+        Self::initialize(storage, private_key)
+    }
+
+    /// Use this to instantiate a PersistentStorage for a new data store, one that has no
+    /// SafetyRules values set.
+    pub fn initialize(
```
Comment (zekun000) [inline consensus/safety-rules/src/persistent_storage.rs]: add some comments/TODO we'll have separate tooling for initialization in prod

**Pull request closed without merging**